pkg/providers/vsphere/session/session_vm_update.go

```diff
@@ -1099,3 +1104,34 @@ func (s *Session) UpdateVirtualMachine(
 
 	return err
 }
+
+func isVMPaused(
+	vmCtx pkgctx.VirtualMachineContext,
+	moVM *mo.VirtualMachine) bool {
+
+	vm := vmCtx.VM
+
+	adminPaused := vmutil.IsPausedByAdmin(moVM)
+
+	var devopsPaused bool
+	if _, ok := vm.Annotations[vmopv1.PauseAnnotation]; ok {
+		devopsPaused = true
+	}
+
+	if vm.Labels == nil {
```

Comment:
Maybe also check `adminPaused || devopsPaused` here since most of the the VMs won't be paused. 

Lots of ways to reduce a few duplicated bool checks, maybe like

```
if adminPaused || devopsPaused {
  if vm.labels == nil {
  }
  switch ... {	
  }
  return true
}
	
delete(vm.Labels, vmopv1.PausedVMLabelKey)
return false
```